cinema/tests/test_movie_api.py

```diff
@@ -157,3 +159,203 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+        self.assertNotIn(serializer_with_actors.data, res.data)
+
+    def test_filter_movie_by_genres(self):
+        movie_without_genres = sample_movie()
+        movie_with_genres_1 = sample_movie(title="test1")
+        movie_with_genres_2 = sample_movie(title="test2")
+
+        genre_1 = sample_genre()
+        genre_2 = sample_genre(name="Comedy")
+
+        movie_with_genres_1.genres.add(genre_1)
+        movie_with_genres_2.genres.add(genre_2)
+
+        res = self.client.get(MOVIE_URL, {"genres": f"{genre_1.id},{genre_2.id}"})
+
+        serializer_without_genres = MovieListSerializer(movie_without_genres)
+        serializer_with_genres_1 = MovieListSerializer(movie_with_genres_1)
+        serializer_with_genres_2 = MovieListSerializer(movie_with_genres_2)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertNotIn(serializer_without_genres.data, res.data)
+        self.assertIn(serializer_with_genres_1.data, res.data)
+        self.assertIn(serializer_with_genres_2.data, res.data)
+
+    def test_filter_movie_by_actors(self):
+        movie_without_actors = sample_movie()
+        movie_with_actors_1 = sample_movie(title="test1")
+        movie_with_actors_2 = sample_movie(title="test2")
+
+        actor_1 = sample_actor()
+        actor_2 = sample_actor()
+
+        movie_with_actors_1.actors.add(actor_1)
+        movie_with_actors_2.actors.add(actor_2)
+
+        res = self.client.get(MOVIE_URL, {"actors": f"{actor_1.id},{actor_2.id}"})
+
+        serializer_without_actors = MovieListSerializer(movie_without_actors)
+        serializer_with_actors_1 = MovieListSerializer(movie_with_actors_1)
+        serializer_with_actors_2 = MovieListSerializer(movie_with_actors_2)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertNotIn(serializer_without_actors.data, res.data)
+        self.assertIn(serializer_with_actors_1.data, res.data)
+        self.assertIn(serializer_with_actors_2.data, res.data)
+
+    def test_retrieve_movie_detail(self):
+        movie = sample_movie()
+        movie.genres.add(sample_genre())
+        movie.actors.add(sample_actor())
+
+        url = detail_url(movie.id)
+
+        res = self.client.get(url)
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        payload = {
+            "title": "test",
+            "description": "test",
+            "duration": 5,
+        }
+
+        res = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class AdminMovieTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="admin@test.com", password="test1234", is_staff=True
+        )
+        self.client.force_authenticate(self.user)
```

Comment:
You can use here 'create_superuser' method, so 'is_staff' parameter won't be necessary to pass in